pkg/libinit/root_linux.go

```diff
@@ -263,24 +320,78 @@ func InstallModules(pattern string, exclude map[string]bool) error {
 	for _, filename := range files {
 		f, err := os.Open(filename)
 		if err != nil {
-			log.Printf("installModules: can't open %q: %v", filename, err)
+			log.Printf("InstallModules: can't open %q: %v", filename, err)
 			continue
 		}
-		// Module flags are passed to the command line in the form modulename.flag=val
+		defer f.Close()
+		// Module flags are passed to the command line in the from modulename.flag=val
 		// And must be passed to FileInit as flag=val to be installed properly
 		moduleName := strings.TrimSuffix(filepath.Base(filename), filepath.Ext(filename))
-		if _, ok := exclude[moduleName]; ok {
-			log.Printf("Skipping module %s", moduleName)
+		if loader.IsExcluded(moduleName) {
+			log.Printf("Skipping module %q", moduleName)
 			continue
 		}
 
 		flags := cmdline.FlagsForModule(moduleName)
-		err = kmodule.FileInit(f, flags, 0)
-		f.Close()
-		if err != nil {
-			log.Printf("installModules: can't install %q: %v", filename, err)
+		if err = kmodule.FileInit(f, flags, 0); err != nil {
+			log.Printf("InstallModules: can't install %q: %v", filename, err)
 		}
 	}
 
 	return nil
 }
+
+func readModules(f *os.File) []string {
+	scanner := bufio.NewScanner(f)
+	modules := []string{}
+	for scanner.Scan() {
+		i := scanner.Text()
+		i = strings.TrimSpace(i)
+		if i == "" {
+			continue
+		} else if strings.HasPrefix(i, "#") {
+			continue
+		} else {
```

Comment:
I think the `else` and `else if` are not needed because all the previous conditions have `continue`.  So you can either do
```
if i == "" {
  continue
}
if strings.HasPrefix(i, "#") {
  continue
}
modules = append(modules, i)
```

or just combine the `if`s:
```
if i = "" || strings.HasPrefix(i, "#") {
  continue
}
modules = append(modules, i)
```

I prefer the latter but don't have a strong preference.